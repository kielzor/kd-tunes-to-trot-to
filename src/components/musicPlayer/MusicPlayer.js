import React from 'react'
// import ReactAudioPlayer from 'react-audio-player'

export const AudioPlayer = () => {
  const playlist = {
    url: '../../tunage/sample-song.mp3',
    cover: '',
    title: 'Sample Song',
    artist: [
      'Kiel Dehnert',
      'Zane Russell',
      'Marcus Spitz'
    ]
  }

  return (
    <div>
      <audio
      controls
      src= "https://freesound.org/people/Freed/sounds/1234">
      </audio>
      {/* <ReactAudioPlayer
        src="http://soundbible.com/mp3/Tyrannosaurus%20Rex%20Roar-SoundBible.com-807702404.mp3"
        autoPlay
        controls
      /> */}
    </div>
  )
}