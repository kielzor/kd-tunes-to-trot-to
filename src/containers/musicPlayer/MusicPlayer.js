import React, { Component } from 'react'
import ReactPlayer from 'react-player'
import { connect } from 'react-redux'

export class MusicPlayer extends Component{
  constructor() {
    super()
    this.state = {
      id: '',
      warmupLength: 0,
      jogLength: 0,
      sprintLength: 0,
      totalLength: 0
    }
  }

  handleChange = e => {
    const { value } = e.target    
    const workoutToPlay = this.props.workout.filter(workout => {
      return workout.name === value
    })
    
    const workout = workoutToPlay[0]

    const jogLength = this.convertLength(workout.jogLength)
    const warmupLength= this.convertLength(workout.warmupLength)
    const sprintLength = this.convertLength(workout.sprintLength)
    const totalLength = this.convertLength(workout.totalLength)

    this.setState ({
      totalLength,
      warmupLength,
      jogLength,
      sprintLength
    })

    this.playWorkout(workout.jog)
  }

  convertLength = time => {
    let length

    if (time === ':30') length = 30000
    if (time === ':45') length = 45000
    if (time === '1:00') length = 60000
    if (time === '1:30') length = 90000
    if (time === '2:00') length = 120000
    if (time === '3:00') length = 180000
    if (time === '5:00') length = 300000
    if (time === '10:00') length = 600000
    if (time === '1:00:00') length = 3600000
    if (time === '1:30:00') length = 5400000
    if (time === '2:00:00') length = 7200000

    return length
  }

  playWorkout = playlist => {
    this.setState ({
      id: `https://freesound.org/apiv2/sounds/${playlist}/download/`
    })
  }

  render() {
    return (
      <div>
        <select
          onChange={this.handleChange}
          value={this.state.value}
        >
          <option selected='selected'>Choose Workout</option>
          {this.props.workout.map(file => {
            return <option>{file.name}</option>
          })}
        </select>
        <ReactPlayer url={this.state.id} playing controls/>
      </div>
    )
  }
}

export const mapStateToProps = state => ({
  workout: state.Workout
})

export default connect(mapStateToProps)(MusicPlayer)