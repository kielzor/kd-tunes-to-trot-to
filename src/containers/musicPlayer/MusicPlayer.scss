.workout-selector {
  width: 40rem;
  margin-top: 2rem;
}

.player-parent {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  background-image: url('https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQ-5v78072GCNPpAdA1sUvbrBdZOV-H2iMCNziQEtpZTH0JlEKlAA');
  background-size: cover;
  height: 80vh;
  opacity: .8;
  color: #2E4756;
}

.player-bottom {
  margin-top: 25rem;
}