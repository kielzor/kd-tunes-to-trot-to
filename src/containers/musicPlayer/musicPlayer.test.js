import React from 'react'
import { shallow } from 'enzyme'
import { MusicPlayer, mapStateToProps } from './MusicPlayer'

describe('MusicPlayer', () => {
	let wrapper

	beforeEach (() => {
		wrapper = shallow(<MusicPlayer workout={[]}/>)
	})
	it('should match the snapshot', () => {
		expect(wrapper).toMatchSnapshot()
	})

	it('should set the state with converted lengths when handleChange is called', () => {
		wrapper = shallow(<MusicPlayer workout={[{ sprintLength: '1:00' }]}/>)
    
		const mockEvent = { target: { sprintLength: { speed: 'Jog', name: 'kiel' } }}    

		wrapper.instance().handleChange(mockEvent)

		expect(wrapper.state('sprintLength')).toEqual(60000)
	})

	it('playJog should set the state correctly', () => {
		wrapper.instance().playJog()

		expect(wrapper.state('currentWorkout')).toEqual('Jog')
	})

	it('playJog should keep the state as Cooldown', () => {
		wrapper.setState({currentWorkout: 'Cooldown'})

		wrapper.instance().playJog()

		expect(wrapper.state('currentWorkout')).toEqual('Cooldown')
	})

	it('playWarmup should set the state correctly', () => {
		wrapper.instance().playWarmup()

		expect(wrapper.state('currentWorkout')).toEqual('Warmup')
	})

	it('playSprint should set the state correctly', () => {
		wrapper.instance().playSprint()

		expect(wrapper.state('currentWorkout')).toEqual('Sprint')
	})

	it('playSprint should keep the state as Cooldown', () => {
		wrapper.setState({currentWorkout: 'Cooldown'})

		wrapper.instance().playSprint()

		expect(wrapper.state('currentWorkout')).toEqual('Cooldown')
	})

	it('playCooldown should set the state to Cooldown', () => {
		wrapper.instance().playCooldown()

		expect(wrapper.state('currentWorkout')).toEqual('Cooldown')
	})

	describe('mapStateToProps', () => {
		it('should return a jogFile array with objects in it', () => {
			const mockState = {
				Workout: [{ name: 'test' }],
			}

			const expected = {
				workout: [{ name: 'test' }],
			}

			const mappedProps = mapStateToProps(mockState)

			expect(mappedProps).toEqual(expected)
		})
	})
})