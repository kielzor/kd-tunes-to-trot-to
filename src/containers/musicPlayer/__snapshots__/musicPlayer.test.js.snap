// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MusicPlayer should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MusicPlayer
    workout={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Choose your workout!!
</h1>,
        <ReactPlayer
          className="player"
          config={
                    Object {
                              "dailymotion": Object {
                                "params": Object {
                                  "api": 1,
                                  "endscreen-enable": false,
                                },
                                "preload": false,
                              },
                              "facebook": Object {
                                "appId": "1309697205772819",
                              },
                              "file": Object {
                                "attributes": Object {},
                                "forceAudio": false,
                                "forceDASH": false,
                                "forceHLS": false,
                                "forceVideo": false,
                                "hlsOptions": Object {},
                                "tracks": Array [],
                              },
                              "mixcloud": Object {
                                "options": Object {
                                  "hide_cover": 1,
                                },
                              },
                              "soundcloud": Object {
                                "options": Object {
                                  "buying": false,
                                  "download": false,
                                  "liking": false,
                                  "sharing": false,
                                  "show_comments": false,
                                  "show_playcount": false,
                                  "visual": true,
                                },
                              },
                              "twitch": Object {
                                "options": Object {},
                              },
                              "vimeo": Object {
                                "playerOptions": Object {
                                  "autopause": false,
                                  "byline": false,
                                  "portrait": false,
                                  "title": false,
                                },
                                "preload": false,
                              },
                              "wistia": Object {
                                "options": Object {},
                              },
                              "youtube": Object {
                                "playerVars": Object {
                                  "iv_load_policy": 3,
                                  "modestbranding": 1,
                                  "playsinline": 1,
                                  "rel": 0,
                                  "showinfo": 0,
                                },
                                "preload": false,
                              },
                            }
          }
          controls={true}
          height="100px"
          loop={true}
          muted={false}
          onBuffer={[Function]}
          onDuration={[Function]}
          onEnded={[Function]}
          onError={[Function]}
          onPause={[Function]}
          onPlay={[Function]}
          onProgress={[Function]}
          onReady={[Function]}
          onSeek={[Function]}
          onStart={[Function]}
          playbackRate={1}
          playing={true}
          playsinline={false}
          progressInterval={1000}
          style={Object {}}
          url=""
          volume={null}
          width="640px"
          wrapper="div"
/>,
        <select
          className="workout-selector"
          onChange={[Function]}
          value={undefined}
>
          <option
                    selected="selected"
          >
                    Choose Workout
          </option>
</select>,
      ],
      "className": "player-parent",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Choose your workout!!",
        },
        "ref": null,
        "rendered": "Choose your workout!!",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "player",
          "config": Object {
            "dailymotion": Object {
              "params": Object {
                "api": 1,
                "endscreen-enable": false,
              },
              "preload": false,
            },
            "facebook": Object {
              "appId": "1309697205772819",
            },
            "file": Object {
              "attributes": Object {},
              "forceAudio": false,
              "forceDASH": false,
              "forceHLS": false,
              "forceVideo": false,
              "hlsOptions": Object {},
              "tracks": Array [],
            },
            "mixcloud": Object {
              "options": Object {
                "hide_cover": 1,
              },
            },
            "soundcloud": Object {
              "options": Object {
                "buying": false,
                "download": false,
                "liking": false,
                "sharing": false,
                "show_comments": false,
                "show_playcount": false,
                "visual": true,
              },
            },
            "twitch": Object {
              "options": Object {},
            },
            "vimeo": Object {
              "playerOptions": Object {
                "autopause": false,
                "byline": false,
                "portrait": false,
                "title": false,
              },
              "preload": false,
            },
            "wistia": Object {
              "options": Object {},
            },
            "youtube": Object {
              "playerVars": Object {
                "iv_load_policy": 3,
                "modestbranding": 1,
                "playsinline": 1,
                "rel": 0,
                "showinfo": 0,
              },
              "preload": false,
            },
          },
          "controls": true,
          "height": "100px",
          "loop": true,
          "muted": false,
          "onBuffer": [Function],
          "onDuration": [Function],
          "onEnded": [Function],
          "onError": [Function],
          "onPause": [Function],
          "onPlay": [Function],
          "onProgress": [Function],
          "onReady": [Function],
          "onSeek": [Function],
          "onStart": [Function],
          "playbackRate": 1,
          "playing": true,
          "playsinline": false,
          "progressInterval": 1000,
          "style": Object {},
          "url": "",
          "volume": null,
          "width": "640px",
          "wrapper": "div",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              selected="selected"
>
              Choose Workout
</option>,
            Array [],
          ],
          "className": "workout-selector",
          "onChange": [Function],
          "value": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Choose Workout",
              "selected": "selected",
            },
            "ref": null,
            "rendered": "Choose Workout",
            "type": "option",
          },
        ],
        "type": "select",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Choose your workout!!
</h1>,
          <ReactPlayer
            className="player"
            config={
                        Object {
                                    "dailymotion": Object {
                                      "params": Object {
                                        "api": 1,
                                        "endscreen-enable": false,
                                      },
                                      "preload": false,
                                    },
                                    "facebook": Object {
                                      "appId": "1309697205772819",
                                    },
                                    "file": Object {
                                      "attributes": Object {},
                                      "forceAudio": false,
                                      "forceDASH": false,
                                      "forceHLS": false,
                                      "forceVideo": false,
                                      "hlsOptions": Object {},
                                      "tracks": Array [],
                                    },
                                    "mixcloud": Object {
                                      "options": Object {
                                        "hide_cover": 1,
                                      },
                                    },
                                    "soundcloud": Object {
                                      "options": Object {
                                        "buying": false,
                                        "download": false,
                                        "liking": false,
                                        "sharing": false,
                                        "show_comments": false,
                                        "show_playcount": false,
                                        "visual": true,
                                      },
                                    },
                                    "twitch": Object {
                                      "options": Object {},
                                    },
                                    "vimeo": Object {
                                      "playerOptions": Object {
                                        "autopause": false,
                                        "byline": false,
                                        "portrait": false,
                                        "title": false,
                                      },
                                      "preload": false,
                                    },
                                    "wistia": Object {
                                      "options": Object {},
                                    },
                                    "youtube": Object {
                                      "playerVars": Object {
                                        "iv_load_policy": 3,
                                        "modestbranding": 1,
                                        "playsinline": 1,
                                        "rel": 0,
                                        "showinfo": 0,
                                      },
                                      "preload": false,
                                    },
                                  }
            }
            controls={true}
            height="100px"
            loop={true}
            muted={false}
            onBuffer={[Function]}
            onDuration={[Function]}
            onEnded={[Function]}
            onError={[Function]}
            onPause={[Function]}
            onPlay={[Function]}
            onProgress={[Function]}
            onReady={[Function]}
            onSeek={[Function]}
            onStart={[Function]}
            playbackRate={1}
            playing={true}
            playsinline={false}
            progressInterval={1000}
            style={Object {}}
            url=""
            volume={null}
            width="640px"
            wrapper="div"
/>,
          <select
            className="workout-selector"
            onChange={[Function]}
            value={undefined}
>
            <option
                        selected="selected"
            >
                        Choose Workout
            </option>
</select>,
        ],
        "className": "player-parent",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Choose your workout!!",
          },
          "ref": null,
          "rendered": "Choose your workout!!",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "player",
            "config": Object {
              "dailymotion": Object {
                "params": Object {
                  "api": 1,
                  "endscreen-enable": false,
                },
                "preload": false,
              },
              "facebook": Object {
                "appId": "1309697205772819",
              },
              "file": Object {
                "attributes": Object {},
                "forceAudio": false,
                "forceDASH": false,
                "forceHLS": false,
                "forceVideo": false,
                "hlsOptions": Object {},
                "tracks": Array [],
              },
              "mixcloud": Object {
                "options": Object {
                  "hide_cover": 1,
                },
              },
              "soundcloud": Object {
                "options": Object {
                  "buying": false,
                  "download": false,
                  "liking": false,
                  "sharing": false,
                  "show_comments": false,
                  "show_playcount": false,
                  "visual": true,
                },
              },
              "twitch": Object {
                "options": Object {},
              },
              "vimeo": Object {
                "playerOptions": Object {
                  "autopause": false,
                  "byline": false,
                  "portrait": false,
                  "title": false,
                },
                "preload": false,
              },
              "wistia": Object {
                "options": Object {},
              },
              "youtube": Object {
                "playerVars": Object {
                  "iv_load_policy": 3,
                  "modestbranding": 1,
                  "playsinline": 1,
                  "rel": 0,
                  "showinfo": 0,
                },
                "preload": false,
              },
            },
            "controls": true,
            "height": "100px",
            "loop": true,
            "muted": false,
            "onBuffer": [Function],
            "onDuration": [Function],
            "onEnded": [Function],
            "onError": [Function],
            "onPause": [Function],
            "onPlay": [Function],
            "onProgress": [Function],
            "onReady": [Function],
            "onSeek": [Function],
            "onStart": [Function],
            "playbackRate": 1,
            "playing": true,
            "playsinline": false,
            "progressInterval": 1000,
            "style": Object {},
            "url": "",
            "volume": null,
            "width": "640px",
            "wrapper": "div",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                selected="selected"
>
                Choose Workout
</option>,
              Array [],
            ],
            "className": "workout-selector",
            "onChange": [Function],
            "value": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Choose Workout",
                "selected": "selected",
              },
              "ref": null,
              "rendered": "Choose Workout",
              "type": "option",
            },
          ],
          "type": "select",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
